src/OrbitGl/CaptureViewElementTest.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include "CaptureViewElementTester.h"
+
+namespace orbit_gl {
+
+class UnitTestCaptureViewElement : public CaptureViewElement {
+ public:
+  explicit UnitTestCaptureViewElement(CaptureViewElement* parent, TimeGraph* time_graph,
+                                      Viewport* viewport, TimeGraphLayout* layout,
+                                      int child_count = 0)
+      : CaptureViewElement(parent, time_graph, viewport, layout) {
+    for (int i = 0; i < child_count; ++i) {
+      children_.emplace_back(
+          std::make_unique<UnitTestCaptureViewElement>(this, time_graph, viewport, layout));
+    }
+  }
+
+  [[nodiscard]] float GetHeight() const override { return 0; }
+  [[nodiscard]] virtual std::vector<CaptureViewElement*> GetChildren() const {
+    std::vector<CaptureViewElement*> result;
+    for (auto& child : children_) {
+      result.push_back(child.get());
+    }
+    return result;
+  };
+
+ private:
+  std::vector<std::unique_ptr<UnitTestCaptureViewElement>> children_;
+
+  [[nodiscard]] std::unique_ptr<orbit_accessibility::AccessibleInterface>
+  CreateAccessibleInterface() override {
+    return nullptr;
+  }
+};
+
+TEST(CaptureViewElementTesterTest, PassesAllTestsOnExistingElement) {
+  CaptureViewElementTester tester;
+  UnitTestCaptureViewElement container_elem(nullptr, nullptr, tester.GetViewport(),
+                                            tester.GetLayout(), 2);
```

Comment:
Why 2?

Reply:
I wanted to make sure changes get propagated to all children, so I kind of arbitrarily choose 2... I wanted it to be more than 1 to catch potential bugs where only the first child is resized.

Reply:
How about using `kArbitraryChildrenCount` here or `kChildrenCount` here instead of directly plugging in the constant 2 in the function? XD

Reply:
Agree with Vicky, wasn't clear for me. BTW, it's impossible to use the real children here?

Reply:
I mean, it is planned in the near future to have the real classes here?

Reply:
@vickyliu-go4it @karupayun Could you take a brief look at the last commit? I've introduced a container and leaf class, hope this makes it a bit clearer. At this point this may be a bit overkill, but I'm thinking of extending this test later.

Reply:
The last commit looks good to me!